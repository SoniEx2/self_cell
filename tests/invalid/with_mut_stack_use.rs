use self_cell::self_cell;

type Dependent<'a> = &'a String;

self_cell!(
    struct MutStackUse {
        #[from_fn]
        owner: String,

        #[covariant]
        dependent: Dependent,
    }
);

fn main() {
    let outside_string = String::from("outside string");

    let mut cell = MutStackUse::from_fn("Crackle that thunder".into(), |owner| owner);

    cell.with_dependent_mut(|_, dependent| {
        *dependent = &outside_string;
    });
}
