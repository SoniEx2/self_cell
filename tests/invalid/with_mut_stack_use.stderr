error[E0597]: `outside_string` does not live long enough
  --> $DIR/with_mut_stack_use.rs:21:23
   |
20 |     cell.with_dependent_mut(|_, dependent| {
   |                             -------------- value captured here
21 |         *dependent = &outside_string;
   |         --------------^^^^^^^^^^^^^^
   |         |             |
   |         |             borrowed value does not live long enough
   |         assignment requires that `outside_string` is borrowed for `'static`
22 |     });
23 | }
   | - `outside_string` dropped here while still borrowed
